************* Module doctest
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1601:0: C0325: Unnecessary parens after 'not' keyword (superfluous-parens)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1609:0: C0325: Unnecessary parens after 'not' keyword (superfluous-parens)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1671:0: C0325: Unnecessary parens after 'not' keyword (superfluous-parens)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2181:0: C0325: Unnecessary parens after 'not' keyword (superfluous-parens)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1:0: C0302: Too many lines in module (2786/1000) (too-many-lines)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:205:4: R1705: Unnecessary "elif" after "return" (no-else-return)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:210:27: W0212: Access to a protected member _getframe of a client class (protected-access)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:225:46: C0103: Variable name "f" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:228:0: C0103: Argument name "s" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:273:4: C0103: Variable name "ws" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:278:4: C0103: Variable name "w" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:285:4: C0103: Variable name "w" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:301:8: C0103: Variable name "w" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:315:4: R1705: Unnecessary "else" after "return" (no-else-return)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:363:20: W0212: Access to a protected member _getframe of a client class (protected-access)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:464:4: R0913: Too many arguments (7/5) (too-many-arguments)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:478:28: C0321: More than one statement on a single line (multiple-statements)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:521:4: R0913: Too many arguments (7/5) (too-many-arguments)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:637:12: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:659:4: R0913: Too many arguments (6/5) (too-many-arguments)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:685:4: C0103: Argument name "m" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:749:12: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:769:4: C0103: Argument name "s" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:772:8: R1705: Unnecessary "else" after "return" (no-else-return)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:777:4: R0201: Method could be a function (no-self-use)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:791:4: R0201: Method could be a function (no-self-use)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:838:4: R0913: Too many arguments (6/5) (too-many-arguments)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:901:51: C0321: More than one statement on a single line (multiple-statements)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:838:4: R0912: Too many branches (19/12) (too-many-branches)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:940:35: W0622: Redefining built-in 'object' (redefined-builtin)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:945:8: R1705: Unnecessary "elif" after "return" (no-else-return)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:940:4: R0201: Method could be a function (no-self-use)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:940:4: R0911: Too many return statements (8/6) (too-many-return-statements)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:968:4: R0913: Too many arguments (8/5) (too-many-arguments)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:968:4: R0912: Too many branches (15/12) (too-many-branches)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1031:4: R0913: Too many arguments (6/5) (too-many-arguments)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1094:34: C0321: More than one statement on a single line (multiple-statements)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1095:36: C0321: More than one statement on a single line (multiple-statements)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1096:37: C0321: More than one statement on a single line (multiple-statements)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1097:33: C0321: More than one statement on a single line (multiple-statements)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1069:4: R0201: Method could be a function (no-self-use)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1069:4: R0912: Too many branches (14/12) (too-many-branches)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:807:0: R0903: Too few public methods (1/2) (too-few-public-methods)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2190:12: C0103: Attribute name "DIVIDER" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1121:0: R0902: Too many instance attributes (12/7) (too-many-instance-attributes)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1217:32: W0613: Unused argument 'test' (unused-argument)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1230:34: W0613: Unused argument 'test' (unused-argument)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1230:40: W0613: Unused argument 'example' (unused-argument)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1230:49: W0613: Unused argument 'got' (unused-argument)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1272:4: R0914: Too many local variables (21/15) (too-many-locals)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1289:8: C0103: Variable name "SUCCESS" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1289:17: C0103: Variable name "FAILURE" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1289:26: C0103: Variable name "BOOM" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1335:12: W0702: No exception type(s) specified (bare-except)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1329:16: W0122: Use of exec (exec-used)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1272:4: R0912: Too many branches (25/12) (too-many-branches)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1272:4: R0915: Too many statements (62/50) (too-many-statements)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1397:4: C0103: Argument name "f" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1397:4: C0103: Argument name "t" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1402:8: C0103: Variable name "f2" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1402:12: C0103: Variable name "t2" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1411:8: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1412:8: R1705: Unnecessary "else" after "return" (no-else-return)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1450:16: E0102: function already defined line 1418 (function-redefined)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1450:16: C0103: Argument name "s" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1485:16: C0415: Import outside toplevel (builtins) (import-outside-toplevel)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1508:12: C0103: Variable name "x" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1509:19: C0103: Variable name "f" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1509:22: C0103: Variable name "t" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1534:24: C0103: Variable name "f" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1534:27: C0103: Variable name "t" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1491:4: R0912: Too many branches (15/12) (too-many-branches)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1549:8: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1550:19: C0103: Variable name "f" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1550:22: C0103: Variable name "t" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1550:28: W0212: Access to a protected member _name2ft of a client class (protected-access)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1556:16: C0103: Variable name "f2" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1556:20: C0103: Variable name "t2" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1557:16: C0103: Variable name "f" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1558:16: C0103: Variable name "t" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1438:8: W0201: Attribute 'test' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1462:8: W0201: Attribute 'debugger' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1468:8: W0201: Attribute 'save_linecache_getlines' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1569:4: C0103: Method name "_toAscii" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1569:4: C0103: Argument name "s" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1569:4: R0201: Method could be a function (no-self-use)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1575:4: R0911: Too many return statements (7/6) (too-many-return-statements)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1638:4: R0201: Method could be a function (no-self-use)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1698:8: R1705: Unnecessary "elif" after "return" (no-else-return)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1718:4: W0231: __init__ method from base class 'Exception' is not called (super-init-not-called)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1737:4: W0231: __init__ method from base class 'Exception' is not called (super-init-not-called)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1837:8: C0103: Variable name "r" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1855:0: C0103: Constant name "master" doesn't conform to UPPER_CASE naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1857:0: C0103: Argument name "m" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1857:0: R0913: Too many arguments (9/5) (too-many-arguments)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1923:4: C0103: Constant name "master" doesn't conform to UPPER_CASE naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1923:4: W0603: Using the global statement (global-statement)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:1961:0: R0913: Too many arguments (12/5) (too-many-arguments)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2041:4: C0103: Constant name "master" doesn't conform to UPPER_CASE naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2041:4: W0603: Using the global statement (global-statement)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2084:0: C0103: Argument name "f" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2084:0: R0913: Too many arguments (6/5) (too-many-arguments)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2111:0: C0103: Constant name "_unittest_reportflags" doesn't conform to UPPER_CASE naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2140:4: C0103: Constant name "_unittest_reportflags" doesn't conform to UPPER_CASE naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2140:4: W0603: Using the global statement (global-statement)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2158:8: C0103: Attribute name "_dt_setUp" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2159:8: C0103: Attribute name "_dt_tearDown" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2151:4: R0913: Too many arguments (6/5) (too-many-arguments)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2175:4: C0103: Method name "runTest" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2191:22: W0612: Unused variable 'tries' (unused-variable)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2332:0: C0103: Function name "DocTestSuite" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2409:0: C0103: Function name "DocFileTest" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2409:0: R0913: Too many arguments (6/5) (too-many-arguments)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2435:0: C0103: Function name "DocFileSuite" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2508:0: C0103: Argument name "s" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2606:0: C0103: Argument name "pm" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2611:0: C0103: Argument name "pm" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2613:4: W0621: Redefining name 'pdb' from outer scope (line 100) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2613:4: W0404: Reimport 'pdb' (imported line 100) (reimported)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2613:4: C0415: Import outside toplevel (pdb) (import-outside-toplevel)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2623:8: W0702: No exception type(s) specified (bare-except)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2622:12: W0122: Use of exec (exec-used)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2625:12: C0103: Variable name "p" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2631:0: C0103: Argument name "pm" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2741:4: C0415: Import outside toplevel (argparse) (import-outside-toplevel)
/log/firmware/firmware/usr/lib/python3.8/doctest.py:2774:12: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)

-----------------------------------
Your code has been rated at 8.64/10

